### ISSUE TITLE
Unable to connect to database on local subnet

### ISSUE THREAD

**Issue by nicklasring** [TablePlus/TablePlus-Linux#37]
1. Database version (Ex: PostgreSQL 10.0):
mysqld  Ver 5.7.26-29 for debian-linux-gnu on x86_64 (Percona Server (GPL), Release '29', Revision '11ad961')

2. TablePlus version (the number on the welcome or about screen, Ex: build 81):
Alpha Build 38

3. Linux distro (Ex: Ubuntu 18.04): 
Debian GNU/Linux 10 (buster), 4.19.0-6-amd64

4. The steps to reproduce this issue:
Create a new connection, enter private IP, username,password, database name

Does not work with either MariaDB or MySQL, tested without setting password, dbname and so on. Unable to connect 'Access denied for user @ ip'

**Comment by dvkndn** [comment]
Hi @nicklasring , the message comes from the database itself, not from TablePlus. Can you double check that it allows you to access (e.g. accessing from another IP)?

Also have you been able to access the database with another client?

**Comment by nicklasring** [comment]
Yes there is no issues connecting to the database from other clients and IP's on the local subnet.

**Comment by pgalilea** [comment]
Same problem here
![tableplus-mysql](https://user-images.githubusercontent.com/1197652/92504809-86ccfe00-f1d9-11ea-9ab4-77b770dc0d7e.png)

I can connect using dbeaver or adminer

**Comment by nicklasring** [comment]
Let's hope it is atleast somewhat usable next patch iteration :)

**Comment by snutij** [comment]
Hi, same results that @pgalilea. 
Works fine with dbeaver ! :)

**Comment by huyphams** [comment]
Are you using the latest release @snutij ?

**Comment by snutij** [comment]
@huyphams, `TablePlus Alpha 0.1 - Build 84` to be exact, seems the latest release

**Comment by beppler** [comment]
It appears that if you use localhost on "Database host/socket" TablePlus try to connect using the socket, if you change to 127.0.0.1 or ::1 it connects using sockets.

**Comment by huyphams** [comment]
Hi @beppler the native library will try to connect using TCP first then if it fails it fallback to the socket.
But if you use the IP, it won't do the fallback.

**Comment by beppler** [comment]
It is weird that when I tried to connect using this information, I received an error mentioning the socket:
![image](https://user-images.githubusercontent.com/66092/201478790-91089d06-35b3-4ee2-b2ba-99430d133b4c.png)

Then when I changed it to use ::1, it worked.

**Comment by zzqayy** [comment]
In linux, when i use mysql-client connect to 127.0.0.1,it's success,but use tableplus,it's failed!
![图片](https://github.com/user-attachments/assets/72e5db5c-f92a-4528-9e22-ce7466cdf737)
![图片](https://github.com/user-attachments/assets/951636e6-7f77-4fa8-bef9-e643fbc6ccbb)
